core/timers/timers.ts

```diff
@@ -203,45 +205,50 @@ export function batchProcessTimers(): void {
 
 export function updateTimers(numberOfCycles: i32): void {
   // Want to increment 4 cycles at a time like an actual GB would
-  let cyclesIncreased: i32 = 0;
+  let cyclesIncreased = 0;
   while (cyclesIncreased < numberOfCycles) {
-    let oldDividerRegister: i32 = Timers.dividerRegister;
+    let oldDividerRegister = Timers.dividerRegister;
+    let curDividerRegister = oldDividerRegister;
     cyclesIncreased += 4;
-    Timers.dividerRegister += 4;
+    curDividerRegister += 4;
 
-    if (Timers.dividerRegister > 0xffff) {
-      Timers.dividerRegister -= 0x10000;
+    if (curDividerRegister > 0xffff) {
+      curDividerRegister -= 0x10000;
     }
 
     if (Timers.timerEnabled) {
+      let timerCounterWasReset = Timers.timerCounterWasReset;
       if (Timers.timerCounterOverflowDelay) {
         Timers.timerCounter = Timers.timerModulo;
         // Fire off timer interrupt
         requestTimerInterrupt();
         Timers.timerCounterOverflowDelay = false;
-        Timers.timerCounterWasReset = true;
-      } else if (Timers.timerCounterWasReset) {
-        Timers.timerCounterWasReset = false;
+        timerCounterWasReset = true;
```

Comment:
Will this also set `Timers.timerCounterWasReset`? As this updates its state, which may be used in other places in the code 😄

Reply:
I update this later: https://github.com/torch2424/wasmboy/pull/265/files#diff-cad28e3e18fc3a0c28632b75d956fccaR230

Reply:
Ahh, I got my mistake) You use `if / else if` instead `if / else`